core/src/main/java/com/graphhopper/util/PathMerger.java

```diff
@@ -99,6 +99,8 @@ public void doWork(PathWrapper altRsp, List<Path> paths, Translation tr) {
                     if (pathIndex + 1 < paths.size()) {
                         ViaInstruction newInstr = new ViaInstruction(fullInstructions.get(fullInstructions.size() - 1));
                         newInstr.setViaCount(pathIndex + 1);
+                        // Similar to the FinishInstruction, ViaInstructions have lenght=0
```

Comment:
We could also enforce this during the Finish Instruction creation [here](https://github.com/graphhopper/graphhopper/blob/091c16e5f78acf9cb32ce2b3e181cac9cae228c5/core/src/main/java/com/graphhopper/routing/InstructionsFromEdges.java#L230)

Reply:
As I understood it, they should have no points. If they have no points, we create intervals like [9,9]. If there is one point, we create an interval [9,10]. If we leave the point, we have to create a workaround for the simplification as well as for the serialization.

Reply:
I just updated the PR to show an alternative approach. Now the Instructions contain one point, but we ignore these points during the simplification and serialization. I think it looks a bit cleaner like this.

Reply:
> If there is one point, we create an interval [9,10]

But that is two points?

> If we leave the point, we have to create a workaround for the simplification as well as for the serialization.

Why? Simplification should be disabled for <3 points.

Reply:
> But that is two points?

No. The second point ref of the interval is equal to the first point of the next interval. Intervals are generated by `[ START_REF, START_REF + POIN_COUNT ]` for all Instructions,
expect the FinshInstruction where we artificially set the point count to 0 in the SimpleRouteSerializer. This PR enforces this behavior for all ViaInstructions and FinishInstructions.

Due to that they don't have their own point in the points array. That's why we have to create the workaround with lenght=0 in the PathSimplification.

Reply:
Hmmh. Let me summarize how it **should** be: every instruction has a first point and a last one, obviously the last point is the first point of the next instruction and the json would be `interval:[first_ref,last_ref]` and `length=last_ref-first_ref`. For Finish and ViaInstruction the result is length=0 (due to first==last) and pointlist.size==1 which makes still sense to me.

Everything which contradicts to this should be fixed e.g. the current ViaInstruction `interval` etc. and/or make the Instructions like the JSON output (and also PathDetails).